src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,341 @@
+    private static int taskCount = 0;
+    private static boolean loaded = false;
+    private static boolean autoSaveMode = false;
+    private static File file = new File(FILE_NAME);
+    private static ArrayList<Task> tasks = new ArrayList<>();
+
+
+
+    public static void main(String[] args) throws IOException {
+        loadFile();
+        System.out.println("Data Directory located: " + FILE_PATH_NAME);
+        welcomeMessage();
+        while (true) {
+            String userCommand = getUserInput();
+            executeCommand(userCommand);
+        }
+    }
+
+
+    private static void autoSave() throws IOException {
+        String msgContent = "";
+        // if file does not exist, create new file
+        int i;
+        for (i = 0; i < taskCount; i++) {
+            msgContent += "\n" + Messages.INDENT + (i + 1) + "." + tasks.get(i);
+        }
+        writeToFile(msgContent);
+    }
+
+    private static void writeToFile(String textToAdd) throws IOException {
+        FileWriter fw = new FileWriter(FILE_NAME, StandardCharsets.UTF_8);
+        fw.write(Messages.FILE_MESSAGE + textToAdd);
+        fw.close();
+    }
+
+    private static void loadFile() throws IOException {
+
+        File theDir = new File("data-folder");
+
+        // create directory if does not exist
+        if (!theDir.exists()) {
+            System.out.println("Creating new directory: " + theDir.getName());
+            boolean result = false;
+            try {
+                theDir.mkdir();
+                result = true;
+            } catch (SecurityException se) {
+                System.out.println("Permission Denited");
+            }
+            if (result) {
+                System.out.println("DIR created");
+            }
+        }
+
+        String content;
+        // Skipping the first 3 lines from file
+        int lines = 0;
+        if (file.createNewFile()) {
+            autoSaveMode = true;
+            loaded = true;
+        } else {
+            autoSaveMode = false;
+            if (!loaded) {
+                //File reading from UTF-8 charset
+                Scanner s = new Scanner(file, StandardCharsets.UTF_8);
+                // loading existing file once and add them into list
+                boolean completed = false;
+                while (s.hasNext()) {
+                    // ignoring the first 3 lines
+                    if (lines < 3 && !completed) {
+                        s.nextLine();
+                    } else {
+                        completed = true;
+                        content = s.nextLine();
+                        int dividePoint1 = content.indexOf(".[");
+                        int dividePoint2 = content.indexOf("] ");
+                        char taskType = content.charAt(dividePoint1 + 2);
+                        String taskDetails = content.substring(dividePoint2 + 2);
+                        String taskIsDone = content.substring(dividePoint1 + 5, dividePoint2);
```

Comment:
Should we apply SLAP here to make the code more readable? Perhaps something like taskType = getTaskType();